erts/emulator/beam/beam_load.c

```diff
@@ -3448,38 +3471,40 @@ gen_put_integer(LoaderState* stp, GenOpArg Fail, GenOpArg Size,
     NATIVE_ENDIAN(Flags);
 	/* Negative size must fail */
     if (Size.type == TAG_i) {
-	op->op = genop_i_new_bs_put_integer_imm_4;
-	op->arity = 4;
-	op->a[0] = Fail;
-	op->a[1].type = TAG_u;
-	if (!safe_mul(Size.val, Unit.val, &op->a[1].val)) {
+        Uint size;
+	if (!safe_mul(Size.val, Unit.val, &size)) {
         error:
             op->op = genop_badarg_1;
             op->arity = 1;
             op->a[0] = Fail;
             op->next = NULL;
             return op;
 	}
-	op->a[1].val = Size.val * Unit.val;
-	op->a[2].type = Flags.type;
-	op->a[2].val = (Flags.val & 7);
-	op->a[3] = Src;
+	op->op = genop_i_new_bs_put_integer_imm_4;
+	op->arity = 4;
+	op->a[0] = Src;
+	op->a[1] = Fail;
+	op->a[2].type = TAG_u;
+	op->a[2].val = size;
+	op->a[3].type = Flags.type;
+	op->a[3].val = (Flags.val & 7);
     } else if (Size.type == TAG_q) {
 	Eterm big = stp->literals[Size.val].term;
 	Uint bigval;
+        Uint size;
 
-	if (!term_to_Uint(big, &bigval)) {
+	if (!term_to_Uint(big, &bigval) ||
+            !safe_mul(bigval, Unit.val, &size)) {
 	    goto error;
-	} else {
-	    op->op = genop_i_new_bs_put_integer_imm_4;
-	    op->arity = 4;
-	    op->a[0] = Fail;
-	    op->a[1].type = TAG_u;
-	    op->a[1].val = bigval * Unit.val;
-	    op->a[2].type = Flags.type;
-	    op->a[2].val = (Flags.val & 7);
-	    op->a[3] = Src;
 	}
+        op->op = genop_i_new_bs_put_integer_imm_4;
+        op->arity = 4;
+        op->a[0] = Fail;
```

Comment:
The argument order doesn't match the changes made in `ops.tab`, so `bs_construct_SUITE` fails to load on 32-bit platforms:

```
2> l(bs_construct_SUITE).    
{error,badfile}
=ERROR REPORT==== 7-Mar-2019::09:11:22.624029 ===
beam/beam_load.c(2309): Error loading function bs_construct_SUITE:opt_dont_call_me/0: op i_new_bs_put_integer_imm p u u i:
  no specific operation found


=ERROR REPORT==== 7-Mar-2019::09:11:22.624218 ===
Loading of /ldisk/daily_build/22_master-opu_o.2019-03-06_21/test/compiler_test/bs_construct_SUITE.beam failed: badfile
```

Reply:
Thanks! Fixed.